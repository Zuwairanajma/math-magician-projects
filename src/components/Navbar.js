/* eslint-disable import/no-extraneous-dependencies */
import { Link } from 'react-router-dom';

function Navbar() {
  return (
    <>
      <nav className="nav-container">
        <h1>Math Magicians</h1>
        <ul className="navbar">
          <li>
            <Link className="nav-item" to="/">Home</Link>
          </li>
          <li>
            <Link className="nav-item" to="/quotes">Quotes</Link>
          </li>
          <li>
            <Link className="nav-item" to="/calculator">Calculator</Link>
          </li>
        </ul>
      </nav>
    </>
  );
}

export default Navbar;
