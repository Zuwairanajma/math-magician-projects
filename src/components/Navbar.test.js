import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/jest-globals';
import Navbar from './Navbar';

test('renders the navbar with correct label', () => {
  render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  );
  const labelElement = screen.getByText('Math Magicians');
  expect(labelElement).toBeInTheDocument();
});
test('renders the navbar with correct links', () => {
  render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  );

  const homeLink = screen.getByRole('link', { name: 'Home' });
  const calculatorLink = screen.getByRole('link', { name: 'Calculator' });
  const quoteLink = screen.getByRole('link', { name: 'Quotes' });

  expect(homeLink).toHaveAttribute('href', '/');
  expect(calculatorLink).toHaveAttribute('href', '/calculator');
  expect(quoteLink).toHaveAttribute('href', '/quotes');
});

test('matches snapshot', () => {
  const { container } = render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  );
  expect(container).toMatchSnapshot();
});
