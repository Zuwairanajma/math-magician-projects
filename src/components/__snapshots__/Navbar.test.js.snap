// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <nav
    class="nav-container"
  >
    <h1>
      Math Magicians
    </h1>
    <ul
      class="navbar"
    >
      <li>
        <a
          class="nav-item"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          class="nav-item"
          href="/quotes"
        >
          Quotes
        </a>
      </li>
      <li>
        <a
          class="nav-item"
          href="/calculator"
        >
          Calculator
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
